api/observation.go

```diff
@@ -50,11 +53,21 @@ func (api *DatasetAPI) getObservations(w http.ResponseWriter, r *http.Request) {
 	version := vars["version"]
 
 	logData := log.Data{"dataset_id": datasetID, "edition": edition, "version": version}
+	auditParams := common.Params{"dataset_id": datasetID, "edition": edition, "version": version}
+
+	if auditErr := api.auditor.Record(r.Context(), getObservationsAction, actionAttempted, auditParams); auditErr != nil {
```

Comment:
If you wanted to write less code, you could set up a go func like:

```go
errorAudit := make(chan bool, 1)
successAudit := make(chan bool, 1)

go func() {
	switch {
	case <-errorAudit:
		if auditErr := api.auditor.Record(r.Context(), getObservationsAction, actionUnsuccessful, auditParams); auditErr != nil {
			handleAuditingFailure(w, auditErr, logData)
			return
		}
	case <-successAudit:
		if auditErr := api.auditor.Record(r.Context(), getObservationsAction, actionSuccessful, auditParams); auditErr != nil {
			handleAuditingFailure(w, auditErr, logData)
			return
		}
	}
}()
```

and then send a message to the relevant channel when the audit is ready to be actioned:

```go
datasetDoc, err := api.dataStore.Backend.GetDataset(datasetID)
if err != nil {
	log.Error(err, logData)
	errorAudit <- true
	handleObservationsErrorType(w, err)
	return
}
```

Reply:
or even:

```go
auditEvent := make(chan string, 1)

go func() {
	action := <-auditEvent
	if auditErr := api.auditor.Record(r.Context(), getObservationsAction, action, auditParams); auditErr != nil {
		handleAuditingFailure(w, auditErr, logData)
		return
	}
}()
```

```go
if err != nil {
	log.Error(err, logData)
	auditEvent <- actionUnsuccessful
	handleObservationsErrorType(w, err)
	return
}
```

Reply:
hmmm, if it fails to audit , which error response is going to be returned; it should be 500 internal server error but i don't think this is guaranteed here. 

I will move on for now